cmake/utils.cmake

```diff
@@ -147,16 +147,23 @@ macro(override_gpu_arches GPU_ARCHES GPU_LANG GPU_SUPPORTED_ARCHES)
   if (${GPU_LANG} STREQUAL "HIP")
     #
     # `GPU_ARCHES` controls the `--offload-arch` flags.
-    # `CMAKE_HIP_ARCHITECTURES` is set up by torch and can be controlled
-    # via the `PYTORCH_ROCM_ARCH` env variable.
     #
-
+    # If PYTORCH_ROCM_ARCH env variable exists, then we take it as a list,
+    # if not, then we use CMAKE_HIP_ARCHITECTURES which was generated by calling
+    # "rocm_agent_enumerator" in "enable_language(HIP)"
+    # (in file Modules/CMakeDetermineHIPCompiler.cmake)
+    #
+    if(DEFINED ENV{PYTORCH_ROCM_ARCH})
```

Comment:
Reusing PYTORCH_ROCM_ARCH environment variable in VLLM environment exposes several potential problems: (1) this environment variable was used by pytorch on rocm when building its rocm backend (https://github.com/pytorch/pytorch/blob/0707811286d1846209676435f4f86f2b4b3d1a17/cmake/Dependencies.cmake#L1075). Reuse this will hide this information, and may potentially make the debugging more difficult. (2) Also, without intersecting with pytorch ROCm targets, users can build any vllm-unsupported rocm target. For example, if a certain user posts a community issue on vllm, and that user used this env to build vllm for gfx900, we can imagine the debugging steps needed for supporting such kind of scenarios.

Reply:
I'm not really sure if this is a concern:
1. This just restores the behavior to what it was prior to https://github.com/vllm-project/vllm/pull/2830
2. We also now install `torch` wheels, so `PYTORCH_ROCM_ARCH` in the base image is decoupled from whatever "Pytorch used when building its ROCm backend"
3. The debugging info provided by this environment variable seems to be very limited.

Reply:
Yes, I am concerned about the name clash. Can you change it to VLLM_ROCM_SUPPORTED_ARCHS?
Though the dockerfile in this PR installed wheels, the wheels was built using the same environment variable and same logic.

Reply:
It was working fine prior to https://github.com/vllm-project/vllm/pull/2830 and the workflow people are used to is to specify `PYTORCH_ROCM_ARCH` to decide the architectures to build vLLM for: this has been the case at ROCm/vllm and here, just that it hasn't been working here in a while.. I haven't yet seen a good reason to change this workflow. If this change is made, it should be made in a separate PR with proper discussion of the change.